Library/Homebrew/unpack_strategy/dmg.rb

```diff
@@ -147,6 +147,12 @@ def extract_to_dir(unpack_dir, basename:, verbose:)
                             verbose:
           end
 
+          # Ditto will try to write as the UID, not the EUID and the Tempfile has 0700 permissions.
+          if Process.euid != Process.uid
```

Comment:
```suggestion
          if bomfile.path.present? && Process.euid != Process.uid
```
should fix your type-checking error. Otherwise seems reasonable to me.

Reply:
Though, maybe not the right fix, since the rest of the code assumes `bomfile.path` is not `nil`.

Reply:
@carlocab thanks, pushed a fix!